stdlib/public/core/Codable.swift

```diff
@@ -5600,6 +5619,14 @@ extension Dictionary: Decodable where Key: Decodable, Value: Decodable {
         let value = try container.decode(Value.self, forKey: key)
         self[key.intValue! as! Key] = value
       }
+    } else if let codingKeyRepresentableType = Key.self as? CodingKeyRepresentable.Type {
+      // The keys are CodingKeyRepresentable, so we should be able to expect a keyed container.
+      let container = try decoder.container(keyedBy: _DictionaryCodingKey.self)
+      for key in container.allKeys {
+        let value = try container.decode(Value.self, forKey: key)
+        let k = codingKeyRepresentableType.init(codingKey: key)
+        self.dict[k as! Key] = value
+      }
```

Comment:
If the key fails to initialize, it might be appropriate to throw an error.

```suggestion
      let container = try decoder.container(keyedBy: _DictionaryCodingKey.self)
      for dictionaryCodingKey in container.allKeys {
        guard let key: Key = codingKeyRepresentableType.init(
          codingKey: dictionaryCodingKey
        ) as? Key else {
          throw DecodingError.dataCorruptedError(
            forKey: dictionaryCodingKey,
            in: container,
            debugDescription: "..."
          )
        }
        let value: Value = try container.decode(
          Value.self,
          forKey: dictionaryCodingKey
        )
        self[key] = value
      }
```

Reply:
You're right - I thought that the types must match up, but missed that the initializer is failable. Thanks!